x/pairing/keeper/pairing.go

```diff
@@ -110,24 +109,28 @@ func (k Keeper) GetPairingForClient(ctx sdk.Context, chainID string, clientAddre
 func (k Keeper) getPairingForClient(ctx sdk.Context, chainID string, clientAddress sdk.AccAddress, block uint64) (providers []epochstoragetypes.StakeEntry, allowedCU uint64, projectID string, errorRet error) {
 	var strictestPolicy planstypes.Policy
 
-	epoch, err := k.VerifyPairingData(ctx, chainID, clientAddress, block)
+	epoch, providersType, err := k.VerifyPairingData(ctx, chainID, clientAddress, block)
 	if err != nil {
 		return nil, 0, "", fmt.Errorf("invalid pairing data: %s", err)
 	}
 
+	possibleProviders, found, epochHash := k.epochStorageKeeper.GetEpochStakeEntries(ctx, epoch, chainID)
+	if !found {
+		return nil, 0, "", fmt.Errorf("did not find providers for pairing: epoch:%d, chainID: %s", block, chainID)
+	}
+
 	project, err := k.GetProjectData(ctx, clientAddress, chainID, block)
 	if err != nil {
 		return nil, 0, "", err
 	}
 
-	strictestPolicy, allowedCU, err = k.getProjectStrictestPolicy(ctx, project, chainID)
+	strictestPolicy, allowedCU, minStake, err := k.getProjectStrictestPolicy(ctx, project, chainID)
```

Comment:
(with reference to other comments on `minStake` - maybe store it inside `strictedPolicy` (and then rid the extra arg to `CalcSlots()` ?)

Reply:
after consulting with @Yaroms, we decided to keep it as it is